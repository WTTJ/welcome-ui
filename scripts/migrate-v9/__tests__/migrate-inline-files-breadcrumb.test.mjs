import { readFileSync } from 'fs'
import { resolve } from 'path'

import { describe, expect, it } from 'vitest'

import { setupMigrationTest } from '../helpers/test-setup.mjs'
import { migrateAll } from '../index.mjs'

describe('Inline Migration - Breadcrumb Components', () => {
  const tempDir = setupMigrationTest('inline-breadcrumb-test', '__fixtures__')

  it('migrates Breadcrumb components with styling props to Breadcrumb with classNames', async () => {
    // Run migration
    await migrateAll(tempDir, { copyDir: false, interactive: false })

    // Read results
    const migratedComponent = readFileSync(resolve(tempDir, 'BreadcrumbComponent.tsx'), 'utf8')

    // Verify with snapshots
    expect(migratedComponent).toMatchSnapshot('breadcrumb-component.tsx')
  })
})
