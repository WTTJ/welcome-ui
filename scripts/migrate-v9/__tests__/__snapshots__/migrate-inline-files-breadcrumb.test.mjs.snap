// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Inline Migration - Breadcrumb Components > migrates Breadcrumb components with styling props to Breadcrumb with classNames > breadcrumb-component.tsx 1`] = `
"import { Breadcrumb } from 'welcome-ui/Breadcrumb'

export const BreadcrumbComponent = () => {
  return (
    <div>
      <Breadcrumb className="mb-lg" separator=">">
        <Breadcrumb.Item className="mb-xl" href="/">
          Introduction
        </Breadcrumb.Item>
        <Breadcrumb.Item className="mb-xl">Disabled</Breadcrumb.Item>
        <Breadcrumb.Item className="mb-xl">Breadcrumb</Breadcrumb.Item>
      </Breadcrumb>
    </div>
  )
}
"
`;
